File: web/src/app/services/pricelist.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

export interface PriceConfigResponse {
  standardPrice: number;
  luxuryPrice: number;
  vanPrice: number;
  kmPrice: number;
  createdAt: string;
}

export interface PriceConfigRequest {
  standardPrice: number;
  luxuryPrice: number;
  vanPrice: number;
  kmPrice: number;
}

@Injectable({
  providedIn: 'root'
})
export class PricelistService {
  private apiUrl = `${environment.apiUrl}/api/v1/pricelist`;

  constructor(private http: HttpClient) {}

  getCurrentPricing(): Observable<PriceConfigResponse> {
    return this.http.get<PriceConfigResponse>(this.apiUrl);
  }

  updatePricing(request: PriceConfigRequest): Observable<void> {
    return this.http.put<void>(this.apiUrl, request);
  }
}

File: mobile/app/src/main/java/com/example/mobile/network/service/PricelistService.java
package com.example.mobile.network.service;

import com.example.mobile.dto.pricelist.PriceConfigRequest;
import com.example.mobile.dto.pricelist.PriceConfigResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.PUT;

public interface PricelistService {

    @GET("/api/v1/pricelist/current")
    Call<PriceConfigResponse> getCurrentPricing();

    @PUT("/api/v1/pricelist/update")
    Call<Void> updatePricing(@Body PriceConfigRequest request);
}